packages/daemon/src/service/trace-manager.ts

```diff
@@ -1,19 +1,59 @@
 import { Transform, TransformCallback } from 'stream';
+import { EventEmitter } from 'events';
 import { open, close, write } from 'fs-extra';
 import { provide, scope, ScopeEnum } from 'midway';
 import { StringDecoder } from 'string_decoder';
-import { generateId } from '@pipcook/pipcook-core';
+import { generateId, PipelineStatus, PluginTypeI } from '@pipcook/pipcook-core';
 
+export type PipcookEventType = 'log' | 'job_status';
+
+export interface JobStatusChangeEvent {
+  jobStatus: PipelineStatus;
+  step?: PluginTypeI;
+  stepAction?: 'start' | 'end';
+}
 /**
  * Log obejct for plugin installing and pipeline running.
  */
-export interface LogObject {
+export class Tracer {
```

Comment:
How about extends from EventEmitter?

Reply:
I think it's not a good idea for parameter type checking.

Reply:
Hmm, it's not related to parameter types, what I just call is to use `super()` instead of `this.emitter`.

Reply:
That will expose `emit`/`on` method, IMO, the relationship between the EventEmitter and Tracer is one of inclusion rather than inheritance.
A `Tracer` object **HAS** an event channel, not **IS**, It also has the log stream.